packages/language-server/src/util.ts

```diff
@@ -502,6 +508,7 @@ export function getCompositeTypeFieldsRecursively(
     fieldTypeNames: Record<string, string>
   },
 ): string[] {
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
   const compositeTypeFieldName = compositeTypeFieldNames.shift()!
```

Comment:
Question: are we sure that `compositeTypeFieldNames` is a non-empty array?

Reply:
this is the current check we have in place `if (compositeTypeFieldNames.length)` which doesn't validate for a non-zero length, so this should just be a case of adding `length > 0`

Reply:
actually I think this is caught anyway 🤔

Reply:
I've updated this to just have an explicit early return in case `compositeTypeFieldName` is undefined as that's effectively what the early return afterwards does